src/settings/NotifTroubleshootingScreen.js

```diff
@@ -0,0 +1,499 @@
+import AlertItem from '../common/AlertItem';
+import ZulipText from '../common/ZulipText';
+import type { Identity } from '../types';
+import type { SubsetProperties } from '../generics';
+import type { ZulipVersion } from '../utils/zulipVersion';
+import { useAppState } from '../reactNativeUtils';
+import * as logging from '../utils/logging';
+import { getHaveServerData } from '../haveServerDataSelectors';
+import { TranslationContext } from '../boot/TranslationProvider';
+import isAppOwnDomain from '../isAppOwnDomain';
+import { openLinkWithUserPreference, openSystemNotificationSettings } from '../utils/openLink';
+import { getOwnUserRole, roleIsAtLeast } from '../permissionSelectors';
+import { Role } from '../api/permissionsTypes';
+
+const {
+  Notifications, // android
+  ZLPNotifications, // ios
+} = NativeModules;
+
+type Props = $ReadOnly<{|
+  navigation: AppNavigationProp<'notif-troubleshooting'>,
+  route: RouteProp<'notif-troubleshooting', void>,
+|}>;
+
+// Keep in sync with Android Notifications.googlePlayServicesAvailability.
+type GooglePlayServicesAvailability = {|
+  +errorCode: number,
+  +errorMessage: string | null, // observed to be null when there was no error
+  +hasResolution: boolean,
+  +isSuccess: boolean,
+|};
+
+/**
+ * Information about system settings, Google services availability, etc.
+ *
+ * A property will be null when we expect a value but don't have it yet.
+ * It'll be undefined if we don't expect a value from the platform. (This
+ * way, it's naturally removed from the JSON report we give the user.)
+ */
+type NativeState = {|
+  +systemSettingsEnabled: boolean | null,
+  +googlePlayServicesAvailability: GooglePlayServicesAvailability | null | void,
+
+  // TODO: …more, e.g.:
+  // TODO(#5484): Android notification sound file missing
+  // TODO(#438): Badge count disabled (once iOS supports it)
+|};
+
+function useNativeState() {
+  const [result, setResult] = React.useState<NativeState>({
+    systemSettingsEnabled: null,
+    googlePlayServicesAvailability: Platform.OS === 'android' ? null : undefined,
+  });
+
+  // Subject to races if calls to a given native method can resolve out of
+  // order (unknown).
+  const getAndSetResult = React.useCallback(async () => {
+    const systemSettingsEnabledPromise: Promise<boolean> =
+      Platform.OS === 'android'
+        ? Notifications.areNotificationsEnabled()
+        : ZLPNotifications.areNotificationsAuthorized();
+
+    systemSettingsEnabledPromise.then(systemSettingsEnabled => {
+      setResult(r => ({ ...r, systemSettingsEnabled }));
+    });
+
+    if (Platform.OS === 'android') {
+      const googlePlayServicesAvailabilityPromise: Promise<GooglePlayServicesAvailability> =
+        Notifications.googlePlayServicesAvailability();
+
+      googlePlayServicesAvailabilityPromise.then(googlePlayServicesAvailability => {
+        setResult(r => ({ ...r, googlePlayServicesAvailability }));
+      });
+    }
+  }, []);
```

Comment:
Here's a way to rewrite this in async/await form:

```suggestion
  const getAndSetResult = React.useCallback(() => {
    (async () => {
      const systemSettingsEnabled: boolean = await (Platform.OS === 'android'
        ? Notifications.areNotificationsEnabled()
        : ZLPNotifications.areNotificationsAuthorized());
      setResult(r => ({ ...r, systemSettingsEnabled }));
    })();

    if (Platform.OS === 'android') {
      (async () => {
        const googlePlayServicesAvailability: GooglePlayServicesAvailability =
          await Notifications.googlePlayServicesAvailability();
        setResult(r => ({ ...r, googlePlayServicesAvailability }));
      })();
    }
  }, []);
```

I think that comes out nicer for the usual reasons that async/await is preferable to the direct Promises API.